language-support/java/codegen/src/main/scala/com/digitalasset/daml/lf/codegen/backend/java/inner/TemplateClass.scala

```diff
@@ -86,10 +88,11 @@ private[inner] object TemplateClass extends StrictLogging {
     ParameterizedTypeName.get(ClassName.get(classOf[Optional[_]]), name)
   private def setOfStrings = ParameterizedTypeName.get(classOf[java.util.Set[_]], classOf[String])
 
-  private def generateContractClass(
+  private[inner] def generateContractClass(
       templateClassName: ClassName,
       key: Option[Type],
       packagePrefixes: Map[PackageId, String],
+      isForInterface: Boolean,
```

Comment:
I would probably change the processing to generate a builder, pass it around and decorate it with what matters for different cases rather than burying the logic deep into this and gating it with a flag.

Reply:
This probably could be a viable solution here to reduce the code duplication between InterfaceClass & TemplateClass :thinking: